hyp3_sdk/jobs.py

```diff
@@ -1,96 +1,221 @@
+        self.expiration_time = expiration_time
+
+    def __repr__(self):
+        return str(self.to_dict())
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+    @staticmethod
+    def from_dict(input_dict: dict):
+        expiration_time = parse_date(input_dict['expiration_time']) if input_dict.get('expiration_time') else None
+        return Job(
+            job_type=input_dict['job_type'],
+            job_id=input_dict['job_id'],
+            request_time=input_dict['request_time'],
+            status_code=input_dict['status_code'],
+            user_id=input_dict['user_id'],
+            name=input_dict.get('name'),
+            job_parameters=input_dict.get('job_parameters'),
+            files=input_dict.get('files'),
+            browse_images=input_dict.get('browse_images'),
+            thumbnail_images=input_dict.get('thumbnail_images'),
+            expiration_time=expiration_time
+        )
+
+    def to_dict(self, for_resubmit: bool = False):
+        job_dict = {
             'job_type': self.job_type,
-            'name': self.job_name,
         }
 
+        for key in ['job_id', 'request_time', 'status_code', 'user_id']:
+            value = self.__getattribute__(key)
+            if not for_resubmit:
+                job_dict[key] = value
+
+        for key in ['name', 'job_parameters']:
+            value = self.__getattribute__(key)
+            if value is not None:
+                job_dict[key] = value
 
-def make_job(job_type: str, job_name: str, job_parameters: dict = {}) -> Job:
-    """Make a generic Job object
+        for key in ['files', 'browse_images', 'thumbnail_images']:
+            value = self.__getattribute__(key)
+            if value is not None and not for_resubmit:
+                job_dict[key] = value
 
-    Args:
-        job_type: The job type
-        job_name: A name for the job (must be <= 20 characters)
-        job_parameters: Extra job parameters specifying custom processing options
+        if self.expiration_time is not None and not for_resubmit:
+            job_dict['expiration_time'] = self.expiration_time.isoformat(timespec='seconds')
 
-    Returns:
-        A Job object
-    """
-    return Job(job_type, job_name, job_parameters)
+        return job_dict
 
+    def complete(self) -> bool:
+        return self.status_code in ('FAILED', 'SUCCEEDED')
 
-def make_autorift_job(job_name: str, granule1: str, granule2: str) -> Job:
-    """Make an autoRIFT Job object
+    def succeeded(self) -> bool:
+        return self.status_code == 'SUCCEEDED'
 
-    Args:
-        job_name: A name for the job (must be <= 20 characters)
-        granule1: The first granule (scene) to use
-        granule2: The second granule (scene) to use
+    def failed(self) -> bool:
+        return self.status_code == 'FAILED'
 
-    Returns:
-        A Job object
-    """
-    return Job('AUTORIFT', job_name, {'granules': [granule1, granule2]})
+    def running(self) -> bool:
```

Comment:
seems duplicative to have both `complete()` and `running()`, the client should just use `if not job.complete()`, or the implementation here should `return not self.complete()`